src/App.js

```diff
@@ -23,21 +49,126 @@ class App extends React.Component {
     }));
   };
 
+  // check for winner using rank
+  // ask MS how to tidy this up
+  checkWinner = () => {
+    if (this.state.currCards.length > 1) {
+      if (this.state.currCards[0].rank > this.state.currCards[1].rank) {
+        this.setState({
+          winnerOfRound: 1,
+          numOfRounds: this.state.numOfRounds - 1,
+          playerOneScore: this.state.playerOneScore + 1,
+          message: "Player 1 wins this round!",
+        });
+      } else if (this.state.currCards[0].rank < this.state.currCards[1].rank) {
+        this.setState({
+          winnerOfRound: 2,
+          numOfRounds: this.state.numOfRounds - 1,
+          playerTwoScore: this.state.playerTwoScore + 1,
+          message: "Player 2 wins this round!",
+        });
+      } else {
+        this.setState({
+          winnerOfRound: 0,
+          numOfRounds: this.state.numOfRounds - 1,
+          message: "It's a tie!",
+        });
+      }
+    } else console.log("cards are not dealt yet.");
+  };
+
+  // react method takes two arguments
+  componentDidUpdate = (prevProp, prevState) => {
+    if (prevState.currCards !== this.state.currCards) {
```

Comment:
yeah, i think this is an acceptable alternative to checking and setting the state for the round winner all at once :+1: